var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -400,7 +400,8 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
     # ###################### Dependencies ##########################
 
     # External Kokkos
-    depends_on("kokkos@4.2.01", when="@15.1.0: +kokkos")
+    depends_on("kokkos@4.3", when="@15.2: +kokkos")
```

Comment:
We could be forward-compatible and do `@4.3:`, and then later constrain it when it becomes known that a newer version is incompatible.

Reply:
Does the kokkos version number get updated when trilinos does? Will @master be broken again at that point unless someone comes in and changes this? Just thinking, either way someone needs to edit this file when versions change.

Reply:
Also, I was wondering if any tests could be added to trilinos to make sure this gets updated when needed.  I am not very familiar with trilinos testing myself.  I will (hopefully) remember this and know how to fix it next time it happens, but it cost me several hours this time.

Reply:
I think it's reasonable to only update the `spack` package for every release and I would guess that's what most packages do anyway.

Reply:
> We could be forward-compatible and do @4.3:, and then later constrain it when it becomes known that a newer version is incompatible.

I think we want to keep it pinned to a specific kokkos version per release range. If not then the concretizer will grab the latest kokkos which the trilinos team may not be ready to adopt.

Reply:
@csiefer2 Does trilinos require even the "patch" version to match?
@psakievich The concretizer will only grab the latest kokkos release with the same *minor* version, not the latest release period. I would hope that if 4.3.00 works, 4.3.01 would as well? or is the kokkos version "continuous" and based on the current date or something?

Reply:
> The concretizer will only grab the latest kokkos release with the same minor version, not the latest release period.

The more you know... I did not know that. I thought it would grab the preferred version by default as long as it was greater than the range supplied.

Reply:
Yeah, `@4.3` will match `4.3.0` and `4.3.1` but not `4.4.0`. The concretizer additionally encodes preferences like "newest version" or "preferred version" but those can't override a hard requirement like a version range.

Reply:
Seriously?? That means you would have to update trilinos any time you want to test a new kokkos bug fix release?

Reply:
And if that's the case then I better revert the change I made...

Reply:
OK thanks for the explanation. I'm reverting my change.